File: count-the-number-of-vowel-strings-in-range.ts
{
    
    function vowelStrings(words: string[], left: number, right: number): number {
        const vowels = new Set(['a', 'e', 'i', 'o', 'u'])
        let c = 0;

        while(left <= right) {
            const word = words[left]

            if (vowels.has(word[0]) && vowels.has(word.slice(-1))) {
                c++
            }

            left++;
        }
        

        return c
    };

    console.log(vowelStrings(["are","amy","u"], 0,2))
}

File: count-the-number-of-vowel-strings-in-range.rs
pub fn vowel_strings(words: Vec<String>, left: i32, right: i32) -> i32 {
    let mut vowels = ['a', 'e', 'i', 'o', 'u'];
    
    let mut count = 0;

    for i in left as usize..=right as usize {
        let size = words[i].len();
        
        let first = words[i].chars().nth(0).unwrap();
        let last = words[i].chars().nth(size - 1).unwrap();

        if vowels.contains(&first) && vowels.contains(&last) {
            count += 1;
        }
    }

    count
}